lib/std/os/windows.zig

```diff
@@ -523,19 +527,186 @@ pub fn ReadFile(in_hFile: HANDLE, buffer: []u8, offset: ?u64, io_mode: std.io.Mo
                 };
                 break :blk &overlapped_data;
             } else null;
-            if (kernel32.ReadFile(in_hFile, buffer.ptr, want_read_count, &amt_read, overlapped) == 0) {
+            var console_mode: DWORD = undefined;
+            const is_console_handle: bool = kernel32.GetConsoleMode(in_hFile, &console_mode) != FALSE;
```

Comment:
Really? You want to call `GetConsoleMode` with every call to `ReadFile`?

Reply:
Some context:
- In https://github.com/ziglang/zig/pull/12400 it used `isTty` which uses `os.isatty` which calls `GetConsoleMode` (so it'd call it on every read/write)
- [Rust uses an `is_console` function](https://github.com/rust-lang/rust/blob/960754090acc9cdd2a5a57586f244c0fc712d26c/library/std/src/sys/windows/stdio.rs#L254-L261) which [is a wrapper around `GetConsoleMode`](https://github.com/rust-lang/rust/blob/960754090acc9cdd2a5a57586f244c0fc712d26c/library/std/src/sys/windows/stdio.rs#L81-L87) (so it calls it on every read/write AFAICT)

However, it seems Go stores the `kind` as `"console"` in newFile to skip the GetConsoleMode on read/write:

- https://github.com/golang/go/blob/6e8caefc19cae465444775f6cd107b138a26cce7/src/os/file_windows.go#L49-L51
- https://github.com/golang/go/blob/6e8caefc19cae465444775f6cd107b138a26cce7/src/internal/poll/fd_windows.go#L419-L420

But still seems to have a `GetConsoleMode` call in a (lower level?) API? I'm not familiar with Go.

- https://github.com/golang/go/blob/6e8caefc19cae465444775f6cd107b138a26cce7/src/runtime/os_windows.go#L637-L645

Reply:
> Really? You want to call `GetConsoleMode` with every call to `ReadFile`?

No, I don't.

However, `std.fs.File` is merely a thin wrapper around the underlying OS file handle. If we want to remember whether the OS handle is a console handle on `std.fs.File` creation, we will need to add OS-specific fields in `std.fs.File` data structure, or add additional records in `std.os.windows` along with a lookup scheme by a `std.fs.File` handle (either pointer or a unique ID), just like POSIX file descriptors.

For minimum intrusive changes with platform-agnostic code (i.e. `std.fs.File`), I chose calling `GetConsoleMode` with every `ReadFile` call. I would like to hear your thoughts on design choices before making any change.